build.gradle

```diff
@@ -42,10 +41,10 @@ task compileEnumGenerator(type: Exec) {
 
```

Comment:
I don't think this will work very well as the task will now be eagerly evaluated during the configuration phase of Gradle, thus it will create the directory which will then be deleted if you do prepend the `clean` task to all of the others, such as `build`.

e.g 
```groovy
task compileEnumGenerator(type: Exec) {
    def libsDir = new File(buildDir, "libs")
    def genSourcesDir = new File(buildDir, "generated-sources")

    doFirst {
        libsDir.mkdirs()
        genSourcesDir.mkdirs()
    }
    ..
}
```

This is applicable for some of the tasks below as well.
```groovy
task generateEnums(type: Exec, dependsOn: compileEnumGenerator) {
    def genSourcesDir = new File(buildDir, "generated-sources/pt/davidafsilva/apple/")
    def authType = new File(genSourcesDir, 'OSXKeychainAuthenticationType.java').absolutePath
    def protocolType =  new File(genSourcesDir, 'OSXKeychainProtocolType.java').absolutePath

    doFirst {
        genSourcesDir.mkdirs()
    }
    ..
```

```groovy
task compileSoLibrary(type: Exec, dependsOn: generateJniHeaders) {
    def libsDir = new File(buildDir, "libs")

    doFirst {
        libsDir.mkdirs()
    }
    ..
```

I can change that later if you don't want to, just pointing it out here so I can double check later as well.